src/main/java/Alan.java

```diff
@@ -0,0 +1,141 @@
+import alanExceptions.AlanException;
+import tasks.TaskList;
+import util.Executor;
+import util.FileParser;
+import util.Storage;
+import util.Ui;
+
+import java.util.Scanner;
+
+/**
+ * This class encapsulates the Chat Bot.
+ */
+public class Alan {
+    public static Alan instance;
+    private Storage alanIO;
+    private final Scanner input;
+    private final Ui ui;
+    private final FileParser fileParser;
+    private final Executor executor;
+    private TaskList taskList;
+
+    /**
+     * Private Constructor.
+     */
+    private Alan() {
+        this.input = new Scanner(System.in);
+        this.ui = new Ui();
+        this.executor = new Executor();
+        this.taskList = new TaskList();
+        this.fileParser = new FileParser();
+
+        try {
+            this.alanIO = new Storage();
+        } catch(AlanException e) {
+            executor.excException(e.getMessage());
+        }
+    }
+
+    /**
+     * Entry creates an Alan singleton
+     *
+     * @param args args.
+     */
+    public static void main(String[] args) {
+        if (Alan.instance == null) {
+            Alan.instance = new Alan();
+        }
+        Alan.instance.start();
+    }
+
+    private void start() {
+        try {
+            this.taskList = new TaskList(fileParser.parseFile(alanIO.read()));
+        } catch (AlanException e) {
+            executor.excException(e.getMessage());
+        }
+        greet();
+        run();
+    }
+
+    private void run() {
+        System.out.println("How may I be of service?");
+
+        label:
+        while (true) {
+            String userInput = input.nextLine();
+            String command = userInput.split(" ", 2)[0];
+
+            try {
+                switch (command) {
+                    case "bye":
```

Comment:
It is good that you avoided deep nesting in this case by using the switch statement 😃